core/block_svg.js

```diff
@@ -704,8 +709,9 @@ Blockly.BlockSvg.prototype.onMouseUp_ = function(e) {
       // Don't throw an object in the trash can if it just got connected.
       this.workspace.trashcan.close();
     }
-  } else if (!this.getParent() && Blockly.selected.isDeletable() &&
-      this.workspace.isDeleteArea(e)) {
+  } else if (deleteArea && !this.getParent() && Blockly.selected.isDeletable()) {
```

Comment:
I keep trying to figure out a way to share some of this logic with the booleans calculated in updateCurosr_, but it seems this is the only line that is actually the same.  Do you think its worth having a wouldDeleteBlock helper?